src/components/Account/AccountTransaction/AccountTxPanel.js

```diff
@@ -0,0 +1,157 @@
+    wif: '',
+    setAmount: noop,
+    setReceiver: noop,
+    setAsset: noop,
+    setStep: noop,
+    doTransfer: noop
+  };
+
+  render() {
+    const { amount, receiver, asset, step } = this.props;
+
+    return (
+      <Panel className={styles.accountPanel} renderHeader={null}>
+        <form className={styles.content}>
+          <h2>Transfer Funds</h2>
+          <div className={styles.inputWrapper}>
+            <Input
+              className={styles.inputs}
+              id="amount"
+              type="number"
+              label="Select Amount"
+              placeholder={0}
+              min={0}
+              step={step}
+              value={amount}
+              disabled={false}
+              onChange={this.handleChangeAmount}
+            />
+            <Select
+              className={styles.inputSelect}
+              defaultValue={asset}
+              onChange={this.handleSelect}
+            >
+              {map(assetsForSelect, (item, index) => (
+                <option value={item.value} key={`asset${index}`}>{item.label}</option>
+              ))}
+            </Select>
+            <Input
+              className={styles.inputs}
+              id="recipient"
+              type="text"
+              label="Transfer to"
+              placeholder="AFnWixEza..."
+              value={receiver}
+              disabled={false}
+              onChange={this.handleChangeReceiver}
+            />
+          </div>
+          <hr />
+          <div className={styles.buttonWrapper}>
+            {/* Future feature - ability to save/add accounts to your address book */}
+            {/* <Button */}
+            {/* className={classNames(styles.buttons, styles.buttonSecondary)} */}
+            {/* type="submit" */}
+            {/* onClick={this.showTransferConfirm} */}
+            {/* > */}
+            {/* <Icon className={styles.Icons} name="add" aria-describedby="add"/> */}
+            {/* Add Account */}
+            {/* </Button> */}
+            <Button className={styles.buttons} type="submit" onClick={this.showTransferConfirm}>
+              <Icon className={styles.Icons} name="transfer" aria-describedby="transfer" />
+              Transfer
+            </Button>
+          </div>
+        </form>
+      </Panel>
+    );
+  }
+
+  handleTransferConfirm = () => {
+    const { doTransfer } = this.props;
+    const { net, asset, amount, receiver, address, wif } = this.props;
+
+    // TODO error handling
+    doTransfer({ net, asset, amount, receiver, address, wif });
+  };
+
+  handleSelect = (event) => {
+    const { value } = event.target;
+
+    if (value === NEO) {
+      this.props.setStep('1');
+    }
+    if (value === GAS) {
+      this.props.setStep('0.00000001');
+    }
+
+    this.props.setAsset(event.target.value);
+  };
+
+  handleChangeAmount = (event) => {
+    const { value } = event.target;
+
+    // Else new BigNumber() fails
+    if (!/^[+-]?([0-9]*[.])?[0-9]+$/.test(value)) return;
```

Comment:
How'd you tackle this problem? @mhuggins 
If a user types "a" or any other letter into the input, new BigNumber(value) throws an error

This might also not be the best regex :P

Reply:
Nvm, this (+ conversion to bignumber) has to be moved just before the transfer - else mutating the state of the value results in the cursor of the input jumping to the end - which is really annoying :P